src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -621,6 +621,10 @@ private function isAllowedProperty(string $class, string $property, bool $writeA
                 return false;
             }
 
+            if (\PHP_VERSION_ID >= 80400 && ($reflectionProperty->isProtectedSet() || $reflectionProperty->isPrivateSet())) {
```

Comment:
`$writeAccessRequired` should be satisfied before checking if setter is protected or private?

Besides, how about using `method_exists` to make Psalm happy?

Reply:
checking for the `PHP_VERSION_ID` constant can be done at compile time, the same is not true when using `method_exists()` instead